script/run_build

```diff
@@ -20,6 +20,9 @@ fi
 
 fold "cukes" run_cukes
 
+# To remove after benchmark measurement
+ruby benchmarks/symbol_to_proc_vs_block.rb
+
```

Comment:
I wouldn't add this to the build personally, whats your thoughts here

Reply:
I think it take some times to gather benchmark result on each Ruby build. You not get easily info that the new Ruby version is for example slower that the previous one on one benchmark. This is more something that should be inserted in https://rubybench.org/ruby/ruby/releases

Also it will add some time to the build itself. I think we should try to have always a fast CI.

I am not convinced about adding it. 😊

I would prefer a benchmark for each gem (except rspec-rails) that assert rspec gem benchmarking performance using a tool like https://github.com/marketplace/actions/continuous-benchmark (it needs Ruby support).

Reply:
It might be an idea to run once, but not for every build.

Reply:
I tried two things:
- Did what Aaron did. Run the benchmark on the CI. But we do not have bare metal server like for https://rubybench.org/hardware. So we cannot have real benchmark. One build can go on a machine that is slower or impacted about installed libraries. I don't think benchmark on CI is a good idea: https://bheisler.github.io/post/benchmarking-in-the-cloud/
- I tried https://github.com/benchmark-driver/benchmark-driver with `bundle exec benchmark-driver -vv --bundler benchmarks/symbol_to_proc_vs_block.rb --rvm '2.2.2;2.3.1;2.4.6;2.5.5;2.6.6;2.7.1'` and:

```ruby
require 'benchmark_driver'

Benchmark.driver do |x|
  x.prelude <<~RUBY
    RANGE = (1..100)

    def sym_to_proc_with_map
      RANGE.map(&:to_s)
    end

    def sym_to_proc_with_each
      RANGE.each(&:to_s)
    end

    def block_with_map
      RANGE.map { |i| i.to_s }
    end

    def block_with_each
      RANGE.each { |i| i.to_s }
    end
  RUBY

  size = 10_000
  range = (1..size)

  x.report 'Block map', %{ block_with_map }
  x.report 'Symbol#to_proc map', %{ sym_to_proc_with_map }
  x.report 'Block each', %{ block_with_each }
  x.report 'Symbol#to_proc each', %{ sym_to_proc_with_each }
end
```

But the gem looks pretty buggy at the moment. I was not able to have the output for all ruby with with this command. Also results are not properly compared at the end you have to do it by hand. Plus, it is working only after Ruby 2.2. 

If you are curious. Here is a cleaned output https://gist.github.com/benoittgt/3464d476060dbbe1dce5e7c649a3587a


BTW I noticed that the part with benchark-ips was not written in the initial benchmark by Aaron. So I rewrote it.

I think both benchmark should be removed. I already removed one. If we want to have benchmarks like this one (basic Ruby accross Ruby version), we should rely on rubybench.